ed25519_test.go

```diff
@@ -70,59 +70,82 @@ func TestMarshalLoop(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	for name, f := range map[string]func() ([]byte, error){
-		"Bytes": priv.Bytes,
-		"Marshal": func() ([]byte, error) {
-			return MarshalPrivateKey(priv)
-		},
-		"NonRedundant": func() ([]byte, error) {
-			// Manually marshal a non-redundant private key to make sure it works!
-			pbmes := new(pb.PrivateKey)
-			pbmes.Type = priv.Type()
-			data, err := priv.Raw()
-			if err != nil {
-				t.Fatal(err)
-			}
-
-			pbmes.Data = data[:ed25519.PrivateKeySize]
-			return pbmes.Marshal()
-		},
-	} {
-		t.Run(name, func(t *testing.T) {
-			bts, err := f()
-			if err != nil {
-				t.Fatal(err)
-			}
-			privNew, err := UnmarshalPrivateKey(bts)
-			if err != nil {
-				t.Fatal(err)
-			}
-
-			if !priv.Equals(privNew) || !privNew.Equals(priv) {
-				t.Fatal("keys are not equal")
-			}
-		})
-	}
-
-	for name, f := range map[string]func() ([]byte, error){
-		"Bytes": pub.Bytes,
-		"Marshal": func() ([]byte, error) {
-			return MarshalPublicKey(pub)
-		},
-	} {
-		t.Run(name, func(t *testing.T) {
-			bts, err := f()
-			if err != nil {
-				t.Fatal(err)
-			}
-			pubNew, err := UnmarshalPublicKey(bts)
-			if err != nil {
-				t.Fatal(err)
-			}
-
-			if !pub.Equals(pubNew) || !pubNew.Equals(pub) {
-				t.Fatal("keys are not equal")
-			}
-		})
-	}
+	t.Run("PrivateKey", func(t *testing.T) {
+		for name, f := range map[string]func() ([]byte, error){
```

Comment:
I also wanted to add a stronger regression test by generating a key with the old code (redundant public key), marshalling it, keeping it hardcoded into the test and verifying that it can still be used.
Do you think it's worth it or is it overkill?